iothub/device/src/InternalClient.cs

```diff
@@ -156,6 +157,8 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
                 Logging.Enter(this, transportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
             }
 
+            JsonConvert.DefaultSettings = () => new JsonSerializerSettings { MaxDepth = 128 };
```

Comment:
Every time this is access, a new settings object will be created. We'd rather access a shared instance.

If we're lucky, Newtonsoft will get it and save the instance, but I wouldn't count on it.

Reply:
https://github.com/JamesNK/Newtonsoft.Json/issues/78#issuecomment-17460093

He made it a delegate intentionally. The case he mentions may not matter to us, though.

Reply:
I updated it to use a shared instance of JsonSerializerSettings. Could you take a look at the implementation again? Thank you in advance!